### PR TITLE
Issue #6916: migrate tests to junit5 for javadoc package

### PR DESCRIPTION
**Pull request by pbludov**
Issue #6916

Tests of the `javadoc` package are migrated to Junit5.

This PR introduces new dependency: https://github.com/itsallcode/junit5-system-extensions
This is replacement for [SystemRules ](https://github.com/stefanbirkner/system-rules)

See changes in AbstractJavadocCheckTest.java

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -278,6 +278,12 @@
       <version>${junit.version}</version>
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>org.itsallcode</groupId>
+      <artifactId>junit5-system-extensions</artifactId>
```
Comment (romani) [inline pom.xml]: > This is replacement for SystemRules 

if it is replacement .... old library should be removed.
please clarify your intend.

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/AbstractJavadocCheckTest.java [hunk 2]
```diff
@@ -129,18 +132,19 @@ public void testWithMultipleChecks() throws Exception {
     }
 
     @Test
-    public void testAntlrError() throws Exception {
+    public void testAntlrError(final Capturable errStream) throws Exception {
```
Comment (romani) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/AbstractJavadocCheckTest.java]: why `final` is required here?

Comment (pbludov) [reply src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/AbstractJavadocCheckTest.java]: Sorry, copypasted from example as is. Removed.

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/AbstractJavadocCheckTest.java [hunk 3]
```diff
@@ -129,18 +132,19 @@ public void testWithMultipleChecks() throws Exception {
...
         };
         verify(checkConfig, getPath("InputAbstractJavadocInvalidAtSeeReference.java"), expected);
-        assertEquals("Error is unexpected", "", systemErr.getLog());
+        assertEquals("", errStream.getCapturedData(), "Error is unexpected");
     }
 
     @Test
-    public void testCheckReuseAfterParseErrorWithFollowingAntlrErrorInTwoFiles() throws Exception {
+    public void testCheckReuseAfterParseErrorWithFollowingAntlrErrorInTwoFiles(
+            final Capturable errStream) throws Exception {
```
Comment (romani) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/AbstractJavadocCheckTest.java]: remove from here too

File: pom.xml (hunk 1 above)
Comment (pbludov) [reply pom.xml]: Yes, `system-rules` dependency will be removed once all tests migrates to junit5

Comment (romani) [reply pom.xml]: please put it somewhere as TODO, it is so easy to forget.

Comment (pbludov) [reply pom.xml]: > please put it somewhere as TODO, it is so easy to forget.

Added to the issue checklist: https://github.com/checkstyle/checkstyle/issues/6916#issuecomment-514506599

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/AbstractJavadocCheckTest.java (hunk 3 above)
Comment (pbludov) [reply src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/AbstractJavadocCheckTest.java]: done

**Pull request merged**